pkg/controller/machine/machine_controller_phases.go

```diff
@@ -241,6 +242,15 @@ func (r *ReconcileMachine) reconcileInfrastructure(ctx context.Context, m *v1alp
 		return &capierrors.RequeueAfterError{RequeueAfter: 30 * time.Second}
 	}
 
+	// Get and set providerID from the infrastructure provider.
+	providerID, _, err := unstructured.NestedString(infraConfig.Object, "spec", "providerID")
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve data from bootstrap provider for Machine %q in namespace %q", m.Name, m.Namespace)
+	} else if providerID == "" {
+		return errors.Errorf("retrieved empty Spec.ProviderID from infrastructure provider for Machine %q in namespace %q", m.Name, m.Namespace)
```

Comment:
Does this turn into a requeue-after?

Reply:
It'll not be a `RequeueAfterError` (the one we define in CAPI), but it'll be requeued given that's gonna be returned to the controller.

Reply:
So that means it will be subject to exponential backoff. I don't know that we want that - it starts at 5ms and doubles each time, up to 1000 seconds. That means that it will quickly try several times, but once it gets to 10 seconds, our backoffs have a much greater impact on the delay - 20s, 40s, 80s, etc. Is that acceptable?

Reply:
It should be fine, in case an infrastructure provider returns Ready=true and an empty ProviderID, I'd consider that an issue with the provider's codebase, wdyt?

Reply:
Thank you GitHub for hiding that crucial detail (this comes after the readiness check) 😄 This is fine.